src/components/map/ha-locations-editor.ts

```diff
@@ -232,6 +243,22 @@ export class HaLocationsEditor extends LitElement {
     fireEvent(this, "marker-clicked", { id: marker.id }, { bubbles: false });
   }
 
+  private _mapClicked(ev) {
+    if (this.moveOnClick && this._locationMarkers) {
+      const id = Object.keys(this._locationMarkers)[0];
+      const location: [number, number] = [
+        ev.detail.location[0],
+        this.normalizeLongitude(ev.detail.location[1]),
+      ];
+      fireEvent(this, "location-updated", { id, location }, { bubbles: false });
+
+      // If the normalized longitude wraps around the globe, pan to the new location.
+      if (location[1] !== ev.detail.location[1]) {
+        this.map.leafletMap?.panTo({ lat: location[0], lng: location[1] });
+      }
```

Comment:
Can you explain what this block is doing? When does it occure?

Reply:
Understood thanks

Reply:
The issue was if you zoom all the way out of the map, you see a continuously wrapping globe, and you can zoom back in to anywhere. So if you zoom into say the right side of the map you could be zoomed into lng = 450 - 470. Then when you click a pin at lng=460, it would get normalized to lng=100, and you wouldn't be able to see it. So it would seem to be broken.